libs/providers/flagsmith/src/lib/flagsmith-provider.ts

```diff
@@ -0,0 +1,198 @@
+        this.events.emit(ProviderEvents.Error);
+      }
+    }
+  }
+
+  /**
+   * Creates a Flagsmith OpenFeature provider.
+   *
+   * @param client - The Flagsmith client instance
+   * @param config - Provider configuration options
+   * @param config.returnValueForDisabledFlags - If true, returns flag values even when disabled. If false, throws error for disabled flags. Default: false
+   * @param config.useFlagsmithDefaults - If true, allows using Flagsmith default flag values. If false, throws FlagNotFoundError for missing flags. Default: false
+   * @param config.useBooleanConfigValue - If true, returns flag.value for boolean flags. If false, returns flag.enabled. Default: false
+   */
+  constructor(client: Flagsmith, config: FlagsmithProviderConfig = {}) {
+    this.client = client;
+    this.returnValueForDisabledFlags = config.returnValueForDisabledFlags ?? false;
+    this.useFlagsmithDefaults = config.useFlagsmithDefaults ?? false;
+    this.useBooleanConfigValue = config.useBooleanConfigValue ?? false;
+  }
+
+  private getFlags(evaluationContext: EvaluationContext): Promise<Flags> {
+    if (evaluationContext?.targetingKey) {
+      return this.client.getIdentityFlags(
+        evaluationContext.targetingKey,
+        (evaluationContext.traits as FlagsmithTrait | undefined) || {},
+      );
+    }
+    return this.client.getEnvironmentFlags();
+  }
+
+  /**
+   * Resolves a feature flag value with type conversion and validation.
+   *
+   * @param flagKey - The feature flag key to resolve
+   * @param flagType - The expected return type ('boolean' | 'string' | 'number' | 'object')
+   * @param evaluationContext - OpenFeature evaluation context with targeting and traits
+   * @returns Promise resolving to the typed flag value
+   * @throws {FlagNotFoundError} When flag doesn't exist and useFlagsmithDefaults is false
+   * @throws {FlagsmithProviderError} When flag is disabled and returnValueForDisabledFlags is false
+   * @throws {TypeMismatchError} When flag value cannot be converted to requested type
+   */
+  private async resolve(
+    flagKey: string,
+    flagType: FlagValueType,
+    evaluationContext: EvaluationContext,
+  ): Promise<ResolutionDetails<any>> {
+    let flag: BaseFlag;
+    try {
+      const flags = await this.getFlags(evaluationContext);
+      flag = flags.getFlag(flagKey);
+    } catch (error) {
+      throw new FlagsmithProviderError('An error occurred retrieving flags from Flagsmith client.', ErrorCode.GENERAL);
+    }
+
+    // Do we want to raise an error if the flag is not found?
+    if (!this.useFlagsmithDefaults && (!flag || flag?.isDefault)) {
+      throw new FlagNotFoundError(`Flag '${flagKey}' was not found.`);
+    }
+
+    if (!(this.returnValueForDisabledFlags || flag.enabled)) {
+      throw new FlagsmithProviderError(`Flag '${flagKey}' is not enabled.`, ErrorCode.GENERAL);
+    }
+
+    if (!this.useBooleanConfigValue && flagType === 'boolean') {
```

Comment:
I changed the priority order, as some tests raised behavior I wouldn't have expected due to conflicting config:
1. Throw error if !flag and !default
2. Throw error if `!returnValueForDisabledFlags` and the flag not enabled
3. At last we `useBooleanConfigValue` (enabled instead of value) => Before it would take priority over 2

Sounds good?

Reply:
No, I looked at this and decided that the order was correct. I think that `useBooleanConfigValue` should take precedent over `returnValueForDisabledFlags`, although I agree these two arguments are not well-defined enough for my liking!